lib/utilities/healthcheckHandler.js

```diff
@@ -45,16 +45,26 @@ export function clientCheck(log, cb) {
 }
 
 function routeHandler(deep, req, res, log, statsClient, cb) {
-    if (isHealthy()) {
-        if (req.method === 'GET' || req.method === 'POST') {
-            if (deep) {
-                return clientCheck(log, cb);
-            }
-            return statsClient.getStats(log, cb);
-        }
+    if (!isHealthy()) {
+        return cb(errors.InternalError, []);
+    }
+    if (req.method !== 'GET' && req.method !== 'POST') {
         return cb(errors.BadRequest, []);
     }
-    return cb(errors.InternalError, []);
+    if (!deep) {
+        return statsClient.getStats(log, cb);
+    }
+    let fail = false;
+    return clientCheck(log, (err, results) => {
+        // makes S3 return 500 error if any backend fails
+        const obj = Object.assign.apply(
+            Object, [{}].concat(results));
```

Comment:
This was my idea. It converts an array of objects into an object.
Originally the intention was to use Object.assign with spread operator 
```js
Object.assign({}, ...arr)
```
but it's not supported. So this is the "babel transpiled" version of that.